nutils/evaluable.py

```diff
@@ -858,7 +858,7 @@ def __iter__(self):
   __pow__ = power
   __abs__ = lambda self: abs(self)
   __mod__  = lambda self, other: mod(self, other)
-  __str__ = __repr__ = lambda self: '{}.{}<{}>'.format(type(self).__module__, type(self).__name__, ','.join(map(str, self.shape)) if hasattr(self, 'shape') else '?')
+  __str__ = __repr__ = lambda self: '{}.{}<{}{}>'.format(type(self).__module__, type(self).__name__, self.dtype.__name__[0] +':' if hasattr(self, 'dtype') else '', ','.join(map(str, self.shape)) if hasattr(self, 'shape') else '?')
```

Comment:
I think I would put the `:` in the format string and put a question mark in the uninitialized state same as for the shape. The fact that both the current and proposed implementation keep the colon visible for scalars (e.g. `Sin<f:>`) is not a bad thing in my opinion, in fact I would propose to use precisely the same for the nodes.